File: mRemoteV1/Forms/OptionsPages/SqlServerPage.vb

Imports mRemote3G.App
Imports mRemote3G.App.Info
Imports mRemote3G.Security

Namespace Forms.OptionsPages
    Public Class SqlServerPage
        Public Overrides Property PageName As String
            Get
                Return Language.Language.strSQLServer.TrimEnd(":")
            End Get
            Set
            End Set
        End Property

        Public Overrides Sub ApplyLanguage()
            MyBase.ApplyLanguage()

            lblExperimental.Text = Language.Language.strExperimental.ToUpper
            lblSQLInfo.Text = Language.Language.strSQLInfo

            chkUseSQLServer.Text = Language.Language.strUseSQLServer
            lblSQLServer.Text = Language.Language.strLabelHostname
            lblSQLDatabaseName.Text = Language.Language.strLabelSQLServerDatabaseName
            lblSQLUsername.Text = Language.Language.strLabelUsername
            lblSQLPassword.Text = Language.Language.strLabelPassword
        End Sub

        Public Overrides Sub LoadSettings()
            MyBase.SaveSettings()

            chkUseSQLServer.Checked = My.Settings.UseSQLServer
            txtSQLServer.Text = My.Settings.SQLHost
            txtSQLDatabaseName.Text = My.Settings.SQLDatabaseName
            txtSQLUsername.Text = My.Settings.SQLUser
            txtSQLPassword.Text = Crypt.Decrypt(My.Settings.SQLPass, General.EncryptionKey)
        End Sub

        Public Overrides Sub SaveSettings()
            MyBase.SaveSettings()

            My.Settings.UseSQLServer = chkUseSQLServer.Checked
            My.Settings.SQLHost = txtSQLServer.Text
            My.Settings.SQLDatabaseName = txtSQLDatabaseName.Text
            My.Settings.SQLUser = txtSQLUsername.Text
            My.Settings.SQLPass = Crypt.Encrypt(txtSQLPassword.Text, General.EncryptionKey)

            Runtime.Startup.DestroySQLUpdateHandlerAndStopTimer()
            frmMain.UsingSqlServer = My.Settings.UseSQLServer
            If My.Settings.UseSQLServer Then
                Runtime.Startup.CreateSQLUpdateHandlerAndStartTimer()
            End If
        End Sub

        Private Sub chkUseSQLServer_CheckedChanged(sender As Object, e As EventArgs) _
            Handles chkUseSQLServer.CheckedChanged
            lblSQLServer.Enabled = chkUseSQLServer.Checked
            lblSQLDatabaseName.Enabled = chkUseSQLServer.Checked
            lblSQLUsername.Enabled = chkUseSQLServer.Checked
            lblSQLPassword.Enabled = chkUseSQLServer.Checked
            txtSQLServer.Enabled = chkUseSQLServer.Checked
            txtSQLDatabaseName.Enabled = chkUseSQLServer.Checked
            txtSQLUsername.Enabled = chkUseSQLServer.Checked
            txtSQLPassword.Enabled = chkUseSQLServer.Checked
        End Sub
    End Class
End Namespace

File: mRemoteV1.ConvertedToC#/Forms/OptionsPages/SqlServerPage.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using mRemoteNG.My;
using mRemoteNG.App;
using mRemoteNG.App.Info;
using mRemoteNG.Security;

namespace mRemoteNG.Forms.OptionsPages
{
	public partial class SqlServerPage
	{
		public override string PageName {
			get { return Language.strSQLServer.TrimEnd(":"); }
			set { }
		}

		public override void ApplyLanguage()
		{
			base.ApplyLanguage();

			lblExperimental.Text = Language.strExperimental.ToUpper();
			lblSQLInfo.Text = Language.strSQLInfo;

			chkUseSQLServer.Text = Language.strUseSQLServer;
			lblSQLServer.Text = Language.strLabelHostname;
			lblSQLDatabaseName.Text = Language.strLabelSQLServerDatabaseName;
			lblSQLUsername.Text = Language.strLabelUsername;
			lblSQLPassword.Text = Language.strLabelPassword;
		}

		public override void LoadSettings()
		{
			base.SaveSettings();

			chkUseSQLServer.Checked = mRemoteNG.My.Settings.UseSQLServer;
			txtSQLServer.Text = mRemoteNG.My.Settings.SQLHost;
			txtSQLDatabaseName.Text = mRemoteNG.My.Settings.SQLDatabaseName;
			txtSQLUsername.Text = mRemoteNG.My.Settings.SQLUser;
			txtSQLPassword.Text = Crypt.Decrypt(mRemoteNG.My.Settings.SQLPass, General.EncryptionKey);
		}

		public override void SaveSettings()
		{
			base.SaveSettings();

			mRemoteNG.My.Settings.UseSQLServer = chkUseSQLServer.Checked;
			mRemoteNG.My.Settings.SQLHost = txtSQLServer.Text;
			mRemoteNG.My.Settings.SQLDatabaseName = txtSQLDatabaseName.Text;
			mRemoteNG.My.Settings.SQLUser = txtSQLUsername.Text;
			mRemoteNG.My.Settings.SQLPass = Crypt.Encrypt(txtSQLPassword.Text, General.EncryptionKey);

			Runtime.Startup.DestroySQLUpdateHandlerAndStopTimer();
			My.MyProject.Forms.frmMain.UsingSqlServer = mRemoteNG.My.Settings.UseSQLServer;
			if (mRemoteNG.My.Settings.UseSQLServer) {
				Runtime.Startup.CreateSQLUpdateHandlerAndStartTimer();
			}
		}

		private void chkUseSQLServer_CheckedChanged(object sender, EventArgs e)
		{
			lblSQLServer.Enabled = chkUseSQLServer.Checked;
			lblSQLDatabaseName.Enabled = chkUseSQLServer.Checked;
			lblSQLUsername.Enabled = chkUseSQLServer.Checked;
			lblSQLPassword.Enabled = chkUseSQLServer.Checked;
			txtSQLServer.Enabled = chkUseSQLServer.Checked;
			txtSQLDatabaseName.Enabled = chkUseSQLServer.Checked;
			txtSQLUsername.Enabled = chkUseSQLServer.Checked;
			txtSQLPassword.Enabled = chkUseSQLServer.Checked;
		}
		public SqlServerPage()
		{
			InitializeComponent();
		}
	}
}
